packages/lib/modules/pool/actions/create/preview/useReclammChart.ts

```diff
@@ -0,0 +1,499 @@
+import { useMemo } from 'react'
+import {
+  calculateLowerMargin,
+  calculateUpperMargin,
+  computeCenteredness,
+} from '@repo/lib/modules/reclamm/reclAmmMath'
+import { bn, fNum } from '@repo/lib/shared/utils/numbers'
+import { useColorMode } from '@chakra-ui/react'
+import { useBreakpoints } from '@repo/lib/shared/hooks/useBreakpoints'
+import { useSelectColor } from '@repo/lib/shared/hooks/useSelectColor'
+
+function getGradientColor(colorStops: string[]) {
+  return {
+    type: 'linear',
+    x: 0,
+    y: 0,
+    x2: 0,
+    y2: 1,
+    colorStops: colorStops.map((color, index) => ({ offset: index, color })),
+  }
+}
+
+// const DEFAULT_PRICE_RATE = '1' // TODO
+
+type UseReclAmmChartParams = {
+  balanceA: number
+  balanceB: number
+  virtualBalanceA: number
+  virtualBalanceB: number
+  margin: number
+  isPriceAdjusting: boolean
+  isPoolWithinTargetRange: boolean
+  isReversed: boolean
+}
+
+export function useReclAmmChart({
+  balanceA,
+  balanceB,
+  virtualBalanceA,
+  virtualBalanceB,
+  margin,
+  isPriceAdjusting,
+  isPoolWithinTargetRange,
+  isReversed,
+}: UseReclAmmChartParams) {
+  const { isMobile } = useBreakpoints()
+  const { colorMode } = useColorMode()
+  const selectColor = useSelectColor()
+
+  const secondaryFontColor = selectColor('font', 'secondary')
+  const highlightFontColor = selectColor('font', 'highlight')
+  const warningFontColor = selectColor('font', 'warning')
+  const borderColor = selectColor('background', 'level0')
+
+  const currentChartData = useMemo(() => {
+    if (!balanceA || !balanceB || !virtualBalanceA || !virtualBalanceB || !margin) {
+      return {}
+    }
+
+    const invariant = bn(bn(balanceA).plus(virtualBalanceA)).times(
+      bn(balanceB).plus(virtualBalanceB)
+    )
+
+    const rBalanceA = balanceA
+    const rBalanceB = balanceB
+    const vBalanceA = virtualBalanceA
+    const vBalanceB = virtualBalanceB
+    const marginValue = margin
+
+    const lowerMargin = calculateLowerMargin({
+      margin: marginValue,
+      invariant: invariant.toNumber(),
+      virtualBalanceA: vBalanceA,
+      virtualBalanceB: vBalanceB,
+    })
+
+    const upperMargin = calculateUpperMargin({
+      margin: marginValue,
+      invariant: invariant.toNumber(),
+      virtualBalanceA: vBalanceA,
+      virtualBalanceB: vBalanceB,
+    })
+
+    const minPriceValue = bn(virtualBalanceB).pow(2).div(invariant).toNumber()
+    const maxPriceValue = bn(invariant).div(bn(virtualBalanceA).pow(2)).toNumber()
+
+    const lowerMarginValue = bn(invariant).div(bn(lowerMargin).pow(2)).toNumber()
+    const upperMarginValue = bn(invariant).div(bn(upperMargin).pow(2)).toNumber()
+
+    const currentPriceValue = bn(bn(balanceB).plus(virtualBalanceB))
+      .div(bn(balanceA).plus(virtualBalanceA))
+      .toNumber()
+
+    // TODO: how to get .priceRate on the pool creation tokens?
+    // OR should we move the whole currentChartData fn out of hook and feed the return values as params?
```

Comment:
@groninge01 this is duplicate reclamm chart based on [ReclAmmChartProvider.tsx](https://github.com/balancer/frontend-monorepo/blob/main/packages/lib/modules/reclamm/ReclAmmChartProvider.tsx)

assuming goal is to share chart btwn pool and creation pages, i think best approach might be to move the entire `currentChartData` fn out of the hook and feed in the return values as params instead. wdyt?

```
    return {
      maxPriceValue,
      minPriceValue,
      lowerMarginValue,
      upperMarginValue,
      currentPriceValue,
      isPoolWithinRange,
      marginValue,
      poolCenteredness,
      isPoolAboveCenter,
    }
```